webapp/src/types/common/index.d.ts

```diff
@@ -12,7 +12,12 @@ type PluginApiService = {
     apiServiceName: ApiServiceName
 }
 
-type PluginState = {
+interface ReduxState extends GlobalState {
+    views: {
+        rhs: {
+            isSidebarOpen: boolean
+        }
+    }
```

Comment:
There is no need for this. You can directly use the GlobalState to access this `views` state.

Reply:
actually, there is no definition for this in MM redux type

```
export declare type GlobalState = {
    entities: {
        general: GeneralState;
        users: UsersState;
        teams: TeamsState;
        channels: ChannelsState;
        posts: PostsState;
        bots: {
            accounts: Dictionary<Bot>;
        };
        preferences: {
            myPreferences: {
                [x: string]: PreferenceType;
            };
        };
        admin: AdminState;
        jobs: JobsState;
        search: SearchState;
        integrations: IntegrationsState;
        files: FilesState;
        emojis: EmojisState;
        typing: Typing;
        roles: {
            roles: {
                [x: string]: Role;
            };
            pending: Set<string>;
        };
        schemes: SchemesState;
        gifs: any;
        groups: GroupsState;
        channelCategories: ChannelCategoriesState;
    };
    errors: Array<any>;
    requests: {
        channels: ChannelsRequestsStatuses;
        general: GeneralRequestsStatuses;
        posts: PostsRequestsStatuses;
        teams: TeamsRequestsStatuses;
        users: UsersRequestsStatuses;
        admin: AdminRequestsStatuses;
        files: FilesRequestsStatuses;
        roles: RolesRequestsStatuses;
        jobs: JobsRequestsStatuses;
    };
    websocket: {
        connected: boolean;
        lastConnectAt: number;
        lastDisconnectAt: number;
    };
};

```

Reply:
Then you should update the version of `mattermost-redux` in `package.json`. @avas27JTG